doc/source/api_changes.rst

```diff
@@ -4,9 +4,181 @@ Release Notes
+  and visualization in a different process or on a different machine.
+
 API Changes
 ^^^^^^^^^^^
 
+* The modules implementing what was loosely dubbed a "SPEC-like" interface
+  (``bluesky.spec_api`` and ``bluesky.global_state``) have been entirely
+  removed. This approach was insufficently similar to SPEC to satisfy SPEC
+  users and confusingly inconsistent with the rest of bluesky.
+
+  The new approach retains the good things about that interface and makes them
+  available for use with *all* plans consistently, including user defined ones.
+  Users who have been fully utilitzing these "SPEC-like" plans will notice four
+  differences.
+
+  1. No ``gs.DETS``. Just use your own variable for detectors. Instead of:
+
+     .. code-block:: python
+
+         # OLD ALTERNATIVE, NO LONGER SUPPORTED
+
+         from bluesky.global_state import gs
+         from bluesky.spec_api import ct
+
+         gs.DETS = # a list of some detectors
+         RE(ct())
+
+     do:
+
+     .. code-block:: python
+
+        from bluesky.plans import count
+
+        dets = # a list of some detectors
+        RE(count(dets))
+
+     Notice that you can use multiple lists to enable easy task switching.
+     Instead of continually updating one global list like this:
+
+     .. code-block:: python
+
+         # OLD ALTERNATIVE, NO LONGER SUPPORTED
+
+         gs.DETS = # some list of detectors
+         RE(ct())
+
+         gs.DETS.remove(some_detector)
+         gs.DETS.append(some_other_detector)
+         RE(ct())
+
+     you can define as many lists as you want and call them whatever you want.
+
+     .. code-block:: python
+
+        d1 = # a list of some detectors
+        d2 = # a list of different detectors
+        RE(count(d1))
+        RE(count(d2))
+
+  2. Automatic baseline readings, concurrent monitoring, and "flying"
+     can be set up uniformly for all plans.
+
+     Formerly, a list of devices to read at the beginning and the end of each
+     run ("baseline" readings), a list of signals to concurrent monitor, and
+     a list of "flyers" to run concurrently were configured like so:
+
+     .. code-block:: python
+
+        # OLD ALTERNATIVE, NO LONGER SUPPORTED
+
+        from bluesky.spec_api import ct
+
+        gs.BASELINE_DEVICES = # a list of devices to read at start and end
+        gs.MONTIORS = # a list of signals to monitor concurrently
+        gs.FLYERS = # a list of "flyable" devices
+
+        gs.DETS = # a list of detectors
+
+        RE(ct())  # monitoring, flying, and baseline readings are added
+
+     And formerly, those settings only affected the behavior of the "SPEC-like"
+     plans, such as ``ct`` and ``ascan``. They were ignored by their
+     counterparts ``count`` and ``scan``, as well as user-defined plans. This
+     was not desirable!
+
+     This scheme has been replaced by the
+     :ref:`diagnostic preprocessor <diagnostic_preprocessor>`, which can be
```

Comment:
That name change would touch a lot of things unrelated to these release notes, so let's duke that out in a separate PR.